core/src/test/java/io/questdb/test/griffin/engine/functions/groupby/LastNotNullGroupByFunctionFactoryTest.java

```diff
@@ -0,0 +1,122 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2023 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.test.griffin.engine.functions.groupby;
+
+import io.questdb.cairo.TableWriter;
+import io.questdb.cairo.sql.Record;
+import io.questdb.cairo.sql.RecordCursor;
+import io.questdb.cairo.sql.RecordCursorFactory;
+import io.questdb.test.AbstractCairoTest;
+import io.questdb.test.tools.TestUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.UUID;
+
+public class LastNotNullGroupByFunctionFactoryTest extends AbstractCairoTest {
+
+    @Test
+    public void testAllNull() throws Exception {
+
+        assertQuery("a0\ta1\ta2\ta3\ta4\ta5\ta6\ta7\ta8\ta9\n" +
+                        "\t\tNaN\tNaN\tNaN\tNaN\t\t\t\t\n",
+                "select last_not_null(a0)a0,last_not_null(a1)a1,last_not_null(a2)a2,last_not_null(a3)a3,last_not_null(a4)a4," +
+                        "last_not_null(a5)a5,last_not_null(a6)a6,last_not_null(a7)a7,last_not_null(a8)a8,last_not_null(a9)a9 from tab",
+                "create table tab as (select cast(null as char)a0,cast(null as date)a1,cast(null as double)a2,cast(null as float)a3,cast(null as int)a4,\n" +
+                        "cast(null as long)a5,cast(null as symbol)a6,cast(null as timestamp)a7,cast(null as uuid)a8,cast(null as string)a9\n" +
+                        "from long_sequence(3))",
+                null,
+                false,
+                true
+        );
+    }
+
```

Comment:
It would be better to use  insert() and assertSql() methods .

Reply:
updated

Reply:
:+1: